installer/cmd/render-mirror.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/docker/distribution/reference"
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+)
+
+type mirrorRepo struct {
+	Original string `json:"original"`
+	Target   string `json:"target"`
+}
+
+var mirrorOpts struct {
+	ExcludeThirdParty bool
+	Output            string
+}
+
+var outputTypes = []string{"yaml", "json"}
+
+// mirrorCmd represents the mirror command
+var mirrorCmd = &cobra.Command{
+	Use:   "mirror",
+	Short: "Renders a list of images used so they can be mirrored to a third-party registry",
+	Long: `Renders a list of images used so they can be mirrored to a third-party registry
+
+A config file is required which can be generated with the init command.
+The "repository" field must be set to your container repository server
+address and this value will be used to generate the mirrored image names
+and tags.
+
+The output can then be used to iterate over each image. A script can
+be written to pull from the "original" path and then tag and push the
+image to the "target" repo`,
+	Example: `
+  # YAML
+  gitpod-installer render mirror --config config.yaml > mirror.yaml
```

Comment:
What's the motivation for the `yaml` format?

Reply:
It's easier to read as a list in the terminal

Reply:
is that worth the variance? How do we test this in the future?